cpu/esp_common/irq_arch_xtensa.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2019 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_esp_common
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the kernels irq interface
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ *
+ * @}
+ */
+
+#if __xtensa__
```

Comment:
Small nitpick: Why not move this to `esp_common/xtensa` and only build the module when `$(CPU_ARCH)` is set to `xtensa`

Reply:
Hm, there is already a module called `xtensa` which refers to the vendor code compiled from `cpu/esp_common/vendor/xtensa`.

But I could imagine to define a new module `arch_xtensa` or `esp_xtensa` which resides in folder `arch_xtensa` or `esp_xtensa` and contains the platform specific files. For RISC-V we could do the same (`arch_riscv` or `arch_esp`).

Reply:
`esp_xtensa` sounds good

Reply:
I think the use of `esp_xtensa` and `esp_riscv` makes it clearer that these are ESP specific modules.